### PR TITLE
napi: Remove code that is not going in PR

### PR DESCRIPTION
**Pull request by sampsongao**
Remove napi_get_current_env(), update affecting code and remove node_api_helpers.h

### CODE REVIEW

File: src/node_jsvmapi.h [hunk 1]
```diff
@@ -22,6 +22,14 @@
...
 #include "node_jsvmapi_types.h"
 
+#define NAPI_METHOD(name)                                                      \
+  void name(napi_env env, napi_callback_info info)
+#define NAPI_GETTER(name) NAPI_METHOD(name)
+#define NAPI_SETTER(name) NAPI_METHOD(name)
+
+#define NAPI_MODULE_INIT(name)                                                 \
+  void name(napi_env env, napi_value exports, napi_value module)
+
```
Comment (mhdawson) [inline src/node_jsvmapi.h]: Did we agree to pull these into the common header ?

Comment (gabrielschulhof) [reply src/node_jsvmapi.h]: IIRC we were gonna drop `NAPI_MODULE_INIT` and `NAPI_METHOD` because the signatures are not supposed to change.

**Review by mhdawson — APPROVED** [review]
LGTM

**Comment by mhdawson** [comment]
CI run: https://ci.nodejs.org/view/x%20-%20Abi%20stable%20module%20API/job/node-test-commit-linux-abi/24/

**Comment by boingoing** [comment]
👍

**Comment by sampsongao** [comment]
Landed in e057a6f974203ad667b460521ced4b818cec4814

**Pull request closed without merging**